build/release.mk

```diff
@@ -117,114 +145,186 @@ endif
+		cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz $(DIST_PATH_LIN_AMD64)/prepackaged_plugins; \
+		cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_AMD64)/prepackaged_plugins; \
+		HAS_ARCH=`tar -tf $(DIST_PATH_LIN_AMD64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz | grep -oE "dist/plugin-.*"`; \
 		if [ "$$HAS_ARCH" != "dist/plugin-linux-amd64" ]; then \
 			echo "Contains $$HAS_ARCH in $$plugin_package-$$ARCH.tar.gz but needs dist/plugin-linux-amd64"; \
 			exit 1; \
 		fi; \
-		gpg --verify $(DIST_PATH_LIN)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz; \
+		gpg --verify $(DIST_PATH_LIN_AMD64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_AMD64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz; \
 		if [ $$? -ne 0 ]; then \
 			echo "Failed to verify $$plugin_package-$$ARCH.tar.gz|$$plugin_package-$$ARCH.tar.gz.sig"; \
 			exit 1; \
 		fi; \
 	done
 	@# Package
-	tar -C $(DIST_PATH_LIN)/.. -czf $(DIST_PATH)-$(BUILD_TYPE_NAME)-linux-amd64.tar.gz mattermost ../mattermost
+	tar -C $(DIST_PATH_LIN_AMD64)/.. -czf $(DIST_PATH)-$(BUILD_TYPE_NAME)-linux-amd64.tar.gz mattermost ../mattermost
+	@# Preserve native package so dev machines will have an unzipped package available
+	@# Cleanup
+	rm -rf $(DIST_ROOT)/linux_amd64
+
+package-linux-arm64: package-prep
+	@# Create needed directories
+	mkdir -p $(DIST_PATH_LIN_ARM64)/bin
+	mkdir -p $(DIST_PATH_LIN_ARM64)/logs
+	mkdir -p $(DIST_PATH_LIN_ARM64)/prepackaged_plugins
+
+	@# Copy binary
+ifeq ($(BUILDER_GOOS_GOARCH),"linux_arm64")
+	cp $(GOBIN)/mattermost $(DIST_PATH_LIN_ARM64)/bin # from native bin dir, not cross-compiled
+else
+	cp $(GOBIN)/linux_arm64/mattermost $(DIST_PATH_LIN_ARM64)/bin # from cross-compiled bin dir
+endif
+	#Download MMCTL for Linux
+	# TODO: add mmctl back once we build arm64
+	# scripts/download_mmctl_release.sh "Linux" $(DIST_PATH_LIN_ARM64)/bin
+	@# Prepackage plugins
+	@for plugin_package in $(PLUGIN_PACKAGES) ; do \
+		ARCH="linux-arm64"; \
+		if test -f "tmpprepackaged/$$plugin_package-$$ARCH.tar.gz"; then \
+			cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz $(DIST_PATH_LIN_ARM64)/prepackaged_plugins; \
+			cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_ARM64)/prepackaged_plugins; \
+			HAS_ARCH=`tar -tf $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz | grep -oE "dist/plugin-.*"`; \
+			if [ "$$HAS_ARCH" != "dist/plugin-linux-arm64" ]; then \
+				echo "Contains $$HAS_ARCH in $$plugin_package-$$ARCH.tar.gz but needs dist/plugin-linux-arm64"; \
+				exit 1; \
+			fi; \
+			gpg --verify $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz; \
+			if [ $$? -ne 0 ]; then \
+				echo "Failed to verify $$plugin_package-$$ARCH.tar.gz|$$plugin_package-$$ARCH.tar.gz.sig"; \
+				exit 1; \
+			fi; \
+		fi; \
```

Comment:
Seems like this is repeated in 3 places now. Wondering if it would be possible to refactor this to a common block?

Reply:
Apologies, I just saw https://github.com/mattermost/mattermost-server/pull/18491#issuecomment-931260977. Any reason we cannot do these refactors with the PR? I'd prefer to keep the Makefile clean if possible.

Reply:
the idea is not to refactor anything now, we are planning to make a better refactor on this soon, and this might disappear, but I need to validate some things before, so I would prefer to keep all the duplication for now